src/recon.rs

```diff
@@ -923,11 +927,25 @@ fn decode_coefs<BD: BitDepth>(
             debug_assert!(x < 32 && y < 32);
             x %= 32;
             y %= 32;
+            // The caller of get_lo_ctx() must ensure that there are at least 65 elements in
+            // level in the TwoD case, and 33 elements in H | V. The size of levels is 32 * 34,
+            // therefore level_off must be such that
+            //  - H|V:  32 * 34 - level_off >= 33 -> level_off <= 1055
+            //  - TwoD: 32 * 34 - level_off >= 65 -> level_off <= 1023
             let level_off = if tx_class == TxClass::TwoD {
+                // TwoD case
+                // rc_i is from scan[i], which has type Scan, which as a max
+                // of 1023 by virtue of the type definition. That's the max from
+                // the calculation above. Therefore we are OK.
                 rc_i as usize
             } else {
+                // H|V case
+                // max x and y are 31, stride = 16
+                // Therefore, max level_off is 527, which is OK.
                 x as usize * stride as usize + y as usize
             };
+            // at this point we know that get_lo_ctx can elide the bounds check on levels
+            // because it is statically known that level has at least 65 elements
```

Comment:
```suggestion
            // At this point, we know that `get_lo_ctx` can elide the bounds check on `level`
            // because it is statically known that `level` has at least 65 elements.
```